### ISSUE TITLE
[Finder] regression with excluding symlinks

### ISSUE THREAD

**Issue by dbu** [symfony/symfony#51314]
### Symfony version(s) affected

5.4.26

### Description

While upgrading from Symfony 5.4.21 to 5.4.26, we noticed that the `assets:install` command no longer installs any assets. We use the `--symlink` and `--relative` flags. With some debugging, i traced it to the [cleanup step](https://github.com/symfony/symfony/blob/3579d245abe4c1840ffe74ac3602268040495d7c/src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php#L176) of `assets:install` removing all the symlinks it just created.

For now we use the workaround to pass `--no-cleanup` to `assets:install`.

### How to reproduce

```
{
    "require": {
        "symfony/finder": "5.4.26",
        "symfony/filesystem": "5.4.21"
    }
}
```

```
<?php
use Symfony\Component\Filesystem\Filesystem;
use Symfony\Component\Finder\Finder;

require ('vendor/autoload.php');

$filesystem = new Filesystem;

$filesystem->remove('public/bundles/mybundle');
$filesystem->mkdir('src/Resources/public/');
$filesystem->mkdir('public/bundles/');
$filesystem->symlink('../../src/Resources/public/', 'public/bundles/mybundle');
$dirsToRemove = Finder::create()->depth(0)->directories()->exclude('mybundle')->in('public/bundles/');
var_dump(iterator_to_array($dirsToRemove));
```

This outputs an entry for the `public/bundles/mybundle` symlink.

With finder 5.4.21, the script outputs an empty array as expected. (composer does not find a 5.4.22-5.4.25)

**Comment by xabbuh** [comment]
Can you please try 5.4.27? This should be fixed with #51178.

**Comment by derrabus** [comment]
Duplicate of #51162

**Comment by dbu** [comment]
thanks! i can confirm that its fixed for me in 5.4.27

sorry for wasting your (and my) time before checking if there already is a new version that fixes the regression :see_no_evil:
